awscli/argprocess.py

```diff
@@ -188,13 +189,20 @@ def _unpack_cli_arg(argument_model, value, cli_name):
         return six.text_type(value)
 
 
-def _unpack_json_cli_arg(argument_model, value, cli_name):
-    try:
-        return json.loads(value, object_pairs_hook=OrderedDict)
-    except ValueError as e:
-        raise ParamError(
-            cli_name, "Invalid JSON: %s\nJSON received: %s"
-            % (e, value))
+def _unpack_json_cli_arg(value, cli_name):
+    if value.lstrip().startswith("---"):
+        try:
+            return yaml.load(value)
```

Comment:
Strongly recommend changing all `yaml.load()` calls to `yaml.safe_load()` instead. For more details, please see https://pyyaml.org/wiki/PyYAMLDocumentation#loading-yaml

While aws-cli is _mostly_ used in situations where the input YAML is safe, it's likely that someone may build a layer on top of aws-cli which takes user input that affects the YAML and might cause a future security bug. Given there's no good reason why you'd want to load arbitrary Python objects, there's no harm done by this change.

Also see: https://github.com/aws/aws-cli/blob/develop/awscli/customizations/cloudformation/yamlhelper.py

Reply:
JSON is YAML. Any compliant YAML parser will be able to parse JSON. No need to implement hacks like looking for the dashes (which is not required to be valid YAML). Just throw out the JSON parser and use the YAML parser to also parse JSON.